pilot/pkg/bootstrap/server_test.go

```diff
@@ -627,6 +630,57 @@ func TestInitOIDC(t *testing.T) {
 	}
 }
 
+func TestWatchDNSCertForK8sCA(t *testing.T) {
+	tests := []struct {
+		name        string
+		certToWatch []byte
+		certRotated bool
+	}{
+		{
+			name:        "expired cert rotation",
+			certToWatch: testcerts.ExpiredServerCert,
+			certRotated: true,
+		},
+		{
+			name:        "valid cert no rotation",
+			certToWatch: testcerts.ServerCert,
+			certRotated: false,
+		},
+	}
+
+	csr := &cert.CertificateSigningRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "abc.xyz",
+		},
+		Status: cert.CertificateSigningRequestStatus{
+			Certificate: testcerts.ServerCert,
+		},
+	}
+	s := &Server{
+		server:                  server.New(),
+		istiodCertBundleWatcher: keycertbundle.NewWatcher(),
+		kubeClient:              kube.NewFakeClient(csr),
+		dnsNames:                []string{"abc.xyz"},
+	}
+
+	for _, tt := range tests {
+		stop := make(chan struct{})
+		t.Run(tt.name, func(t *testing.T) {
+			s.istiodCertBundleWatcher.SetAndNotify(testcerts.ServerKey, tt.certToWatch, testcerts.CACert)
+
+			go s.watchDNSCertForK8sCA(stop, "", "test-signer", true, time.Duration(0))
+			sleep.Until(stop, 1*time.Second)
+			close(stop)
+
+			st := string(s.istiodCertBundleWatcher.GetKeyCertBundle().CertPem)
+			certRotated := st != string(tt.certToWatch)
+			if certRotated != tt.certRotated {
+				t.Fatalf("expect certRotated is %v while actual certRotated is %v", tt.certRotated, certRotated)
+			}
```

Comment:
nit: wondering if for the expired cert-test case, could we add a few more validations on the returned cert? for instance, just ensure that the new cert has a valid lifetime?

Reply:
Done.